svm-decode/src/lib.rs

```diff
@@ -83,23 +83,83 @@ pub fn svm_decode_instructions(
     batch: &RecordBatch,
     allow_decode_fail: bool,
 ) -> Result<RecordBatch> {
-    let data_col = batch.column_by_name("data").unwrap();
+    let data_col = batch
+        .column_by_name("data")
+        .context("data column not found in instructions batch")
+        .unwrap();
     let data_array = data_col.as_any().downcast_ref::<BinaryArray>().unwrap();
 
-    let account_arrays: Vec<&BinaryArray> = (0..10)
-        .map(|i| {
-            let col_name = format!("a{}", i);
-            let col = batch.column_by_name(&col_name).unwrap();
-            col.as_any().downcast_ref::<BinaryArray>().unwrap()
-        })
-        .collect();
+    let mut account_arrays: Vec<Option<BinaryArray>> = Vec::with_capacity(10);
+
+    for i in 0..10 {
+        let col_name = format!("a{}", i);
+        if let Some(col) = batch.column_by_name(&col_name) {
+            if let Some(binary_array) = col.as_any().downcast_ref::<BinaryArray>() {
+                account_arrays.push(Some(binary_array.clone()));
+            } else {
+                account_arrays.push(None);
+            }
+        }
+    }
+
+    if signature.accounts_names.len() > 10 {
+        let rest_of_acc_col = batch
+            .column_by_name("rest_of_accounts")
+            .context("rest_of_accounts column not found in instructions batch")
+            .unwrap();
+        let rest_of_acc_arrays: &GenericListArray<i32> = rest_of_acc_col
+            .as_any()
+            .downcast_ref::<GenericListArray<i32>>()
+            .context("unable to downcast rest_of_accounts to a list array")
+            .unwrap();
+        // Unpack the rest_of_accounts list array into individual arrays
+        let data_size = rest_of_acc_arrays.len() * 32;
+        for i in 10..signature.accounts_names.len() {
+            let mut builder = BinaryBuilder::with_capacity(rest_of_acc_arrays.len(), data_size);
+            if signature.accounts_names[i].is_empty() {
+                for _ in 0..rest_of_acc_arrays.len() {
+                    builder.append_null();
+                }
+                account_arrays.push(Some(builder.finish()));
+                continue;
+            }
```

Comment:
Couldn't figure out a better way of allowing users to skip accounts they don't care about other than that.